syft/pkg/cataloger/binary/cataloger.go

```diff
@@ -0,0 +1,78 @@
+package binary
+
+import (
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/source"
+)
+
+const catalogerName = "binary-cataloger"
+
+func NewBinaryCataloger() pkg.Cataloger {
```

Comment:
as a rule of thumb we should return the concretion not the abstraction
```suggestion
func NewBinaryCataloger() *Cataloger {
```